File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/lits/domain/LitsObject.java
package com.zwstudio.logicpuzzlesandroid.puzzles.lits.domain;

public abstract class LitsObject {
    public abstract String objAsString();
    public static LitsObject objFromString(String str) {
        switch (str) {
            case "marker":
                return new LitsMarkerObject();
            case "tree":
                return new LitsTreeObject();
            default:
                return new LitsEmptyObject();
        }
    }
}

File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/lits/domain/LitsObject.kt
package com.zwstudio.logicpuzzlesandroid.puzzles.lits.domain

abstract class LitsObject {
    abstract fun objAsString(): String

    companion object {
        fun objFromString(str: String?): LitsObject {
            return when (str) {
                "marker" -> LitsMarkerObject()
                "tree" -> LitsTreeObject()
                else -> LitsEmptyObject()
            }
        }
    }
}